routes/truckRoutes.js

```diff
@@ -1,3 +1,77 @@
 // connect to mongodb in this module as this is where you'll be making creat/read/delete calls to your database
 // use 'mongodb://localhost/foodTruckAPI' for your mongoose connection string
 // remember this is a Node module
+var express = require('express');
+
+var bodyparser = require('body-parser');
+var router = express.Router();
+
+var Truck = require('../models/truckModel');
+
+router.use(bodyparser.urlencoded({ extended: true}));
+router.use(bodyparser.json());
+router.use('/:id', function (req, res, next) {
+
+	Truck.findById(req.params.id, function (error, truck) {
+		if (error) {
+			res.status(500).send(error);
+		} else if (truck) {
+			req.truck = truck;
+			next();
+		} else {
+			res.status(404).send('truck not found');
+		}
+	});
+});
+
+router.route('/')
+	.get(function (req, res) {
+		
+		Truck.find(function (error, trucks) {
+			if (error) {
+				res.status(500).send(error);
+			} else {
+				res.json(trucks);
+			}
+		});
+	})
+	.post(function (req, res) {
+
+		var truck = new Truck(req.body);
+		if (truck) {
+			truck.save();
+			res.status(201).send(truck);

```

Comment:
I would do this call in a callback to the save() call. Better to indicate the new resource was created once the database call is done so that if there's an error, it can be handled with an appropriate response.